include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -289,6 +331,15 @@ class CombinationalArithBinaryLibraryOp<string mnemonic> :
   let results = (outs AnyType:$left, AnyType:$right, AnyType:$out);
 }
 
+class ArithBinaryFloatingPointLibraryOp<string mnemonic> : ArithBinaryLibraryOp<mnemonic, [
+  SameTypeConstraint<"left", "out">
+  ]> {
+    let results = (outs I1:$clk, I1:$reset, I1:$go, I1:$control, I1:$subOp,
+      AnyFloat:$left, AnyFloat:$right, AnySignlessInteger:$roundingMode, AnyFloat:$out,
+      AnySignlessInteger:$execptionalFlags, I1:$done);
+}
+
+def AddFNOp : ArithBinaryFloatingPointLibraryOp<"addFN"> {}
```

Comment:
What is the `addFN` name? "Add floating number" or something like that? I don't usually nitpick on the mnemonics, but it is worth thinking through. I get the impression this aligns with Calyx native IR, so if that's the case, `addFN` sounds good to me.